docs/modules/ROOT/pages/account-abstraction.adoc

```diff
@@ -0,0 +1,194 @@
+
+===== A note on upgradeability
+
+xref:upgradeable.adoc[Upgradeable Contracts] might easily violate ERC-7562 storage access rules during the validation phase. For example, when upgradeability is present in a module (i.e. an external validator), the account will need to call the proxy and access the implementation address in storage.
+
+The general recommendation for upgradeable accounts is to use a xref:account-abstraction.adoc#Signer[`Signer`] so that the account accesses its own storage during the validation phase, which is allowed by the ERC-7562 rules and preserves the ability of the account of executing operations permissionlessly.
+
+=== Signers
+
+An Account controlled by a signer is a type of account that depends on a https://en.wikipedia.org/wiki/Digital_signature[digital signature] verification algorithm and is similar to Externally Owned Accounts (EOAs) in that they require a private key signature to authenticate it. These accounts can be used with ERC-4337 infrastructure normally and do not violate xref:account-abstraction.adoc#erc_7562_validation_rules[ERC-7562 Validation Rules] when used to validate signatures externally in their upgradeable version.
+
+Generally speaking, blockchains use the properties of https://en.wikipedia.org/wiki/Digital_signature[digital signatures] to authenticate the sender of each transaction. With this in mind, developers can xref:utilities.adoc#checking_signatures_on_chain[verify signatures on-chain] as a way to authenticate user operations for a given off-chain signer (i.e. a private key).
+
+To setup an account signer, start by inheriting from `AccountSigner` and xref:api:account.adoc#ERC1271TypedSigner[`ERC1271TypedSigner`]:
+
+```solidity
+include::api:example$account/MyAccountECDSATypedSigner.sol[]
+```
+
+As you would note, the Account requires to implement both `_validateUserOp` and `_validateSignature`. These are 2 different functions because xref:api:account.adoc#ERC1271TypedSigner[`ERC1271TypedSigner`] adds a wrapper to the `userOpHash` and any other hash received through the `isValidSignature` function the account exposes. The wrapper makes sure that the signature isn't replayable across different account implementations and that the user saw the types associated to the signed message.
+
+IMPORTANT: Accounts must be deployed via an xref:account-abstraction.adoc#account_factory[account factory].
+
+==== Readable Typed Signatures
+
+A common security practice to prevent user operation https://mirror.xyz/curiousapple.eth/pFqAdW2LiJ-6S4sg_u1z08k4vK6BCJ33LcyXpnNb8yU[replayability across smart contract accounts controlled by the same private key] (i.e. multiple accounts for the same signer) is to link the signature to the `address` and `chainId`. This can be done by asking the user to sign the hash of the user operation along with these values.
```

Comment:
```suggestion
A common security practice to prevent user operation https://mirror.xyz/curiousapple.eth/pFqAdW2LiJ-6S4sg_u1z08k4vK6BCJ33LcyXpnNb8yU[replayability across smart contract accounts controlled by the same private key] (i.e. multiple accounts for the same signer) is to link the signature to the `address` and `chainId` of the account. This can be done by asking the user to sign the hash of the user operation along with these values.
```